### ISSUE TITLE
Performance discrepancy in ELSI tests

### ISSUE THREAD

**Issue by ocaisa** [easybuilders/easybuild-easyconfigs#14172]
In https://github.com/easybuilders/easybuild-easyconfigs/pull/14133, I saw a large discrepancy between the test times for `foss` and `intel` (~2 minutes compared to ~30 minutes).

Tagging @micaeljtoliveira

**Comment by migueldiascosta** [comment]
@ocaisa, fwiw,  I get (on an AMD Naples node) 1 min 18 secs for `foss` and 1 min 39 secs for `intel`, so the problem seems to be system dependent?

**Comment by migueldiascosta** [comment]
indeed, on Haswell it also takes too long for me. So it's not just the threads, it also depends on the cpu (?)

**Comment by migueldiascosta** [comment]
one difference I see is that on AMD all the processes and threads are using the same socket, while on Intel not only the processes span both sockets, which isn't a problem, but threads with the same parent process span different sockets, which could explain the performance difference

These tests are not using any explicit affinity setting, so I suppose it's either OpenMPI or OpenBLAS that's setting different threading affinities in Intel and AMD?

On the other hand, if that was the case, surely this would have been noticed before (?)

curiouser and curiouser...
